File: server/src/entity/Product.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity({ name: 'product' })
export class Product {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ type: 'varchar', length: 50, nullable: false })
  name: string;

  @Column({ type: 'text', array: true, nullable: false })
  sizes: string[];

  @Column({ type: 'text', array: true, nullable: false })
  colors: string[];

  @Column({ type: 'text', array: true, nullable: false })
  images: string[];

  @Column({ type: 'numeric', precision: 6, scale: 2, nullable: false })
  price: string;

  @Column({ type: 'int', nullable: false })
  quantity: number;

  @Column({ type: 'text', nullable: false })
  description: string;

  @Column({ type: 'varchar', length: 50, nullable: false, unique: true })
  slug: string;

  @Column({ type: 'numeric', precision: 6, scale: 2, default: null })
  discount_price: number;

  @Column({
    type: 'varchar',
    length: 20,
    enum: ['accessories', 'footwear', 'tshirts', 'pants'],
  })
  collection: 'accessories' | 'footwear' | 'tshirts' | 'pants';
}

File: API/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace API.Entities
{
	public class Product
	{
		public Guid Id { get; set; } = Guid.NewGuid();
		public string Name { get; set; }
		public string Slug { get; set; }
		public decimal Price { get; set; }
		public string Description { get; set; }
		public List<string> Images { get; set; }
		public List<string> Colors { get; set; }
		public List<string> Sizes { get; set; }
		public int Quantity { get; set; }
		public Collection Collection { get; set; }
		public int CollectionId { get; set; }
	}
}
